internal/gitutil/git_test.go

```diff
@@ -0,0 +1,464 @@
+	if err != nil {
+		return err
+	}
+	// add upstream remote
+	if _, err := r.CreateRemote(&config.RemoteConfig{
+		Name: "upstream",
+		URLs: []string{newCloneRepo},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// TestEnsureCloned tests the EnsureCloned function
+func TestEnsureCloned(t *testing.T) {
+	// create a local git repository in testdata
+	if err := createGitRepo(t, testRemoteRepo); err != nil {
+		t.Fatalf("failed to create test repo: %v", err)
+	}
+	type args struct {
+		uri             string
+		destinationPath string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "test clone existing repo",
+			args: args{
+				uri:             testRemoteRepo,
+				destinationPath: testClonedRepo,
+			},
+			wantErr: false,
+		},
+		{
+			name: "test where uri is invalid",
+			args: args{
+				uri:             testUnknownRepo,
+				destinationPath: testEmptyRepo,
+			},
+			wantErr: true,
+		},
+		{
+			name: "test where destination path is invalid",
+			args: args{
+				uri:             testUnknownRepo,
+				destinationPath: testEmptyRepo,
+			},
+			wantErr: true,
+		},
+		{
+			name: "test where repo is not cloned and not a git repo",
+			args: args{
+				uri:             testEmptyRepo,
+				destinationPath: testUnknownRepo,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := EnsureCloned(tt.args.uri, tt.args.destinationPath); (err != nil) != tt.wantErr {
+				t.Errorf("EnsureCloned() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+	// cleanup the git testdata repositories used for the testing in this package.
+	t.Run("cleanup test git repos", func(t *testing.T) {
+		t.Cleanup(func() {
+			os.RemoveAll(testRemoteRepo)
+			os.RemoveAll(testClonedRepo)
+		})
+	})
+}
+
+// TestIsGitCloned tests IsGitCloned function
+func TestIsGitCloned(t *testing.T) {
+	// setup local git repositories in testdata
+	setupRemoteRepos(t)
+	type args struct {
+		gitPath string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "test if the test-cloned-repo is cloned",
+			args: args{
+				gitPath: testClonedRepo,
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "test check for unknown repo",
+			args: args{
+				gitPath: testUnknownRepo,
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "test check for git repo",
+			args: args{
+				gitPath: testEmptyRepo,
+			},
+			want:    false,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := IsGitCloned(tt.args.gitPath)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("IsGitCloned() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("IsGitCloned() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+	// cleanup the git testdata repositories used for the testing in this package.
```

Comment:
ditto here

you can solve all these by just calling a `setupTempDirs(t)` that creates dirs and returns a base tmp path, and registers cleanup methods that's executed on each run.

Reply:
Thank you very much !! 
Let me address this with the suggested approach.